src/goto-instrument/code_contracts.cpp

```diff
@@ -168,23 +168,30 @@ void code_contractst::add_quantified_variable(
   replace_symbolt &replace,
   irep_idt mode)
 {
-  // If the expression is a quantified expression, this function adds
-  // the quantified variable to the symbol table and to the expression map
-
-  // TODO Currently only works if the contract contains only a single
-  // quantified formula
-  // i.e. (1) the top-level element is a quantifier formula
-  // and (2) there are no inner quantifier formulae
-  // This TODO is handled in PR #5968
-
-  if(expression.id() == ID_exists || expression.id() == ID_forall)
-  {
-    // get quantified symbol
-    exprt tuple = expression.operands().front();
-    exprt quantified_variable = tuple.operands().front();
+  // This function recursively searches the expression to find nested or
+  // non-nested quantified expressions. When a quantified expression is found,
+  // the quantified variable is added to the symbol table
+  // and to the expression map.
+  if(
+    expression.id() == ID_and || expression.id() == ID_or ||
+    expression.id() == ID_implies)
```

Comment:
What about negations? And the ternary `?:` operator?

Could you please also add regression tests for them?